content/blog/2015-06-03-service-discovery-introduction.md

```diff
@@ -0,0 +1,328 @@
+to query targets.
+
+The tags of each Consul node are concatenated by a configurable separator and exposed 
+through the `__meta_consul_tags` label. Various other meta labels are extracted.
+
+Scraping all instances for a list of given services can be achieved with a simple
+`consul_sd_config` and relabeling rules:
+
+```
+scrape_configs:
+- job_name: 'overwritten-default'
+
+  consul_sd_configs:
+  - server:     '127.0.0.1:5361'
+
+    services: ['auth', 'api', 'load-balancer', 'postgres']
+
+  relabel_configs:
+  - source_labels: ['__meta_consul_service']
+    regex:         '(.*)'
+    target_label:  'job'
+    replacement:   '$1'
+  - source_labels: ['__meta_consul_node']
+    regex:         '(.*)'
+    target_label:  'instance'
+    replacement:   '$1'
+  - source_labels: ['__meta_consul_tags']
+    regex:         '(production|canary)'
+    target_label:  'group'
+    replacement:   '$1'
+```
+
+This synchronizes the given services from the local Consul agent.
+As a result, we get metrics for four jobs (`auth`, `api`, `load-balancer`, and `postgres`). If a node
+has the `production` or `canary` tag a respective `group` label is assigned to the target. 
+Each target's `instance` label is set to the node name provided by Consul.
+
+A full documentation of all configuration parameters for service discovery via Consul
+can be found on the [Prometheus website](/docs/operating/configuration##relabeling-relabel_config).
+
+
+## Custom service discovery
+
+Finally, we added a file-based interface to talk to your custom service discovery or other common mechanisms, 
+that are not yet supported out-of-the-box.
+
+Prometheus watches a set of directories or files which contain target group information. Whenever
+any of those files changes, a list of target groups is read from the files and scrape targets 
+are extracted. 
+It's now our job to write a small bridge program that runs as Prometheus's side-kick. 
+It retrieves changes from a service discovery mechanism and writes the target information
+to the watched files as lists of target groups.
+
+The files can either be in YAML...
+
+```
+- targets: ['10.11.150.1:7870', '10.11.150.4:7870']
+  labels:
+    job: 'mysql'
+
+- targets: ['10.11.122.11:6001', '10.11.122.15:6002']
+  labels:
+    job: 'postgres'
+```
+
+... or JSON format.
+
+```
+[
+  {
+    "targets": ["10.11.150.1:7870", "10.11.150.4:7870"],
+    "labels": {
+      "job": "mysql"
+    }
+  },
+  {
+    "targets": ["10.11.122.11:6001", "10.11.122.15:6002"],
+    "labels": {
+      "job": "postgres"
+    }
+  }
+]
+```
+
+We now configure Prometheus to watch the `tgroups/` folder in its working directory 
+for all `.json` files:
+
+```
+scrape_configs:
+- job_name: 'overwritten-default'
+
+  file_sd_configs:
+  - names: ['tgroups/*.json']
+```
+
+What's missing now is a program that writes files to the directory. For the sake of this example, 
+let's assume we have all our instances for different jobs in one large unnormalized MySQL table.
```

Comment:
Too much unnecessary detail. "in one large unnormalized MySQL table" -> "in a single MySQL table"